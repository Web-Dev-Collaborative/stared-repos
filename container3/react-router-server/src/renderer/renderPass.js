import React from 'react';
import { renderToString, renderToStaticMarkup } from 'react-dom/server';
import AsyncRenderer from '../components/AsyncRenderer';
import removeDuplicateModules from '../utils/removeDuplicateModules';

const renderPass = (context, element, staticMarkup = false) => {
  context.callback = () => {
    if (context.finishedLoadingModules && !context.statesRenderPass) {
      context.statesRenderPass = true;
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    } else if (context.finishedLoadingModules && context.statesRenderPass || !context.hasModules) {
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    }
  };

  let component = (
    <AsyncRenderer context={context}>
      {element}
    </AsyncRenderer>
  );

  let result;
  try {
    if (staticMarkup) {
      result = renderToStaticMarkup(component);
    } else {
      result = renderToString(component);
    }
  } catch (e) {
    return context.reject(e);
  }

  if (!context.hasModules && !context.hasStates) {
    context.resolve({
      html: result,
      state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
      modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
    });
  }
  return result;
};

export default renderPass;
