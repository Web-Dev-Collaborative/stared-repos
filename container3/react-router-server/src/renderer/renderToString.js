import renderPass from './renderPass';
import { clear } from '../module/cache';

export default (element) => {
  return new Promise((resolve, reject) => {
    clear();
    const context = {
      resolve, reject,
      modulesLoading: 0,
      fetchingStates: 0
    };
    renderPass(context, element);
  });
};
