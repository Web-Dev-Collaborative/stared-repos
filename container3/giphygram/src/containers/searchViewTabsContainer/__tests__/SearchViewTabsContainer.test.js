import React from 'react';
import renderer from 'react-test-renderer';
import SearchViewTabsContainerConnected, { SearchViewTabsContainer } from '../SearchViewTabsContainer';

jest.mock('../../../components/searchViewTabs/SearchViewTabs', () => ({
  SearchViewTabs: 'SearchViewTabs',
}));

jest.mock('react-redux', () => ({
  connect: mapStateToProps => (Component) => {
    const state = {
      layout: {
        size: 'sm',
      },
      searchResults: {
        data: [],
        pagination: {
          total_count: 100,
        },
        meta: {},
        isLoading: false,
        isFetchingMore: false,
        error: null,
      },
    };
    const propsFromState = mapStateToProps(state);
    return props => <Component {...props} {...propsFromState} />;
  },
}));

describe('SearchViewTabsContainer', () => {
  it('should be rendered correctly by default', () => {
    const setColumnsNum = jest.fn();

    const tree = renderer
      .create((
        <SearchViewTabsContainer
          isHidden={false}
          setColumnsNum={setColumnsNum}
        />
      ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be rendered correctly with specified columns number', () => {
    const setColumnsNum = jest.fn();

    const tree = renderer
      .create((
        <SearchViewTabsContainer
          isHidden={false}
          setColumnsNum={setColumnsNum}
          columnsNum={4}
        />
      ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be hidden is it is required by properties', () => {
    const setColumnsNum = jest.fn();

    const tree = renderer
      .create((
        <SearchViewTabsContainer
          isHidden
          setColumnsNum={setColumnsNum}
        />
      ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be rendered and connected correctly by default', () => {
    const setColumnsNum = jest.fn();

    const tree = renderer
      .create((
        <SearchViewTabsContainerConnected
          setColumnsNum={setColumnsNum}
        />
      ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should fire onColumnsNumChange callback', () => {
    const setColumnsNum = jest.fn();

    const searchViewTabsContainerComponent = renderer
      .create((
        <SearchViewTabsContainer
          isHidden={false}
          setColumnsNum={setColumnsNum}
        />
      )).root;

    const searchViewTabsComponent = searchViewTabsContainerComponent.findByType('SearchViewTabs');

    expect(searchViewTabsComponent).toBeDefined();

    // Emulate onColumnsNumChange event.
    searchViewTabsComponent.props.onColumnsNumChange(4);

    expect(setColumnsNum).toHaveBeenCalledTimes(1);
    expect(setColumnsNum).toHaveBeenCalledWith(4);
  });
});
