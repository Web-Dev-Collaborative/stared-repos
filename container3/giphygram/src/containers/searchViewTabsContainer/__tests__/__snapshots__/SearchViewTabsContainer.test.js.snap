// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchViewTabsContainer should be hidden is it is required by properties 1`] = `null`;

exports[`SearchViewTabsContainer should be rendered and connected correctly by default 1`] = `
<SearchViewTabs
  columnsNum={3}
  onColumnsNumChange={[Function]}
/>
`;

exports[`SearchViewTabsContainer should be rendered correctly by default 1`] = `
<SearchViewTabs
  columnsNum={3}
  onColumnsNumChange={[Function]}
/>
`;

exports[`SearchViewTabsContainer should be rendered correctly with specified columns number 1`] = `
<SearchViewTabs
  columnsNum={4}
  onColumnsNumChange={[Function]}
/>
`;
